modules/frontend/searchsharding.go

```diff
@@ -188,43 +189,43 @@ func (s searchSharder) RoundTrip(r *http.Request) (*http.Response, error) {
 
 			// if not found send it down the pipeline
 			if !foundInCache {
-				resp, err := s.next.RoundTrip(innerR)
-				if err != nil {
+				resp, rtErr := s.next.RoundTrip(innerR)
+				if rtErr != nil {
 					// context cancelled error happens when we exit early.
 					// bail, and don't log and don't set this error.
-					if errors.Is(err, context.Canceled) {
-						_ = level.Debug(s.logger).Log("msg", "exiting early from sharded query", "url", innerR.RequestURI, "err", err)
+					if errors.Is(rtErr, context.Canceled) {
+						_ = level.Debug(s.logger).Log("msg", "exiting early from sharded query", "url", innerR.RequestURI, "err", rtErr)
 						return
 					}
 
-					_ = level.Error(s.logger).Log("msg", "error executing sharded query", "url", innerR.RequestURI, "err", err)
-					progress.setError(err)
+					_ = level.Error(s.logger).Log("msg", "error executing sharded query", "url", innerR.RequestURI, "err", rtErr)
+					progress.setError(rtErr)
 					return
 				}
 
 				// if the status code is anything but happy, save the error and pass it down the line
 				if resp.StatusCode != http.StatusOK {
 					statusCode := resp.StatusCode
-					bytesMsg, err := io.ReadAll(resp.Body)
-					if err != nil {
-						_ = level.Error(s.logger).Log("msg", "error reading response body status != ok", "url", innerR.RequestURI, "err", err)
+					bytesMsg, ioErr := io.ReadAll(resp.Body)
```

Comment:
any particular reason you split these errors?

Reply:
This routine err was shadowing the err outside the routine, so I think the change make it more clear.